### PR TITLE
CFD-249: Field widgets : Group category st level should not be selectable

### PR DESCRIPTION
**Pull request by HelenaEksler**
#386

### CODE REVIEW

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/directives/group-categories/group-categories.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                var childID = child.id;
+                if (childID in scope.model && scope.model[childID] === true) {
+                  scope.model[childID] = false;
+                }
+              });
+            }
+          });
+
+          if (selectedCount > 3) {
+            angular.element("#" + scope.type + "_description").css(
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/directives/group-categories/group-categories.js]: would be nicer to assign a class, and not hardcode the color

**Comment by HelenaEksler** [comment]
categories

**Comment by HelenaEksler** [comment]
@amitaibu PR is ready for the code review.

File: capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module [hunk 2]
```diff
@@ -211,6 +207,23 @@ function c4m_og_vocab_field_widget_form(&$form, &$form_state, $field, $instance,
   }
   $settings['c4m']['data']['tags'] = $tags;
 
+  if (isset($entity->og_vocabulary) && !empty($entity->og_vocabulary)) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/og/c4m_og_vocab/c4m_og_vocab.module]: enough just ``!empty($entity->og_vocabulary)``

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/main.js [hunk 3]
```diff
@@ -162,6 +162,23 @@ angular.module('c4mApp')
       QuickPostService.togglePopover(name, event, $scope.popups);
     };
 
+    /**
+     * Check if current category has at least ont selected child.
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/main.js]: ont => one

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/main.js [hunk 4]
```diff
@@ -162,6 +162,23 @@ angular.module('c4mApp')
...
+     *
+     * @param key
+     *  Category term id.
+     *
+     * @returns {boolean}
+     */
+    $scope.categoryHasChildrenSelected = function(key) {
+      var result = false;
+      angular.forEach($scope.categories[key].children, function(child, childKey) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/controllers/main.js]: maybe it's better to have a ``for`` loop here, so you could return early as you find the first ``true``

File: capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/directives/group-categories/group-categories.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                var childID = child.id;
+                if (childID in scope.model && scope.model[childID] === true) {
+                  scope.model[childID] = false;
+                }
+              });
+            }
+          });
+
+          if (selectedCount > 3) {
+            angular.element("#" + scope.type + "_description").addClass('tooMany');
```
Comment (amitaibu) [inline capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/directives/group-categories/group-categories.js]: mayeb ``tooMany`` => ``error-too-many-selected`` ?

**Comment by HelenaEksler** [comment]
@amitaibu Fixed. Can I assign it to Peter?

**Pull request merged**